paperqa/agents/search.py

```diff
@@ -436,15 +437,14 @@ async def get_directory_index(
     Args:
         index_name: Deprecated override on the name of the index. If unspecified,
             the default behavior is to generate the name from the input settings.
-        sync_index_w_directory: Sync the index (add or delete index files) with the
-            source paper directory.
+        sync_index_w_directory: Opt-out flag to sync the index (add or delete index
+            files) with the source paper directory.
         settings: Application settings.
+        build: Opt-out flag (default is True) to read the contents of the source paper
```

Comment:
IMO this logic is a bit complicated for users— what do you think about a “from_directory” class method on the SearchIndex class that you can use when you don’t need to build one? Then you can call that instead of adding an extra flag to this function. Logic looks good otherwise though.

Reply:
Looks like you were approximately a few seconds too slow on the PR comment haha.

I didn't decompose `get_directory_index` in this PR because I was trying to be backwards compatible. I will make a downstream PR that does this then